src/harness/sas_objects.py

```diff
@@ -0,0 +1,387 @@
+        """
+        Args:
+            cbsd_id:  cbsd id of the device extracted from the registration response
+            registration_request: A dictionary with a single key-value pair where the key is
+            "registrationRequest" and the value is a registration request content
+            grant_ids: list of grant ids
+            grant_requests: A list of dictionary with a single key-value pair where the key is
+                "grantRequest" and the value is a list of individual CBSD grant
+                requests (each of which is itself a dictionary).
+        """
+        assert (len(grant_requests) == len(grant_ids))
+        self.cbsd_id = cbsd_id
+        self.grant_objects = {}
+        index = 0
+        for grant_request in grant_requests:
+            self.grant_objects[grant_ids[index]] = Grant(
+                grant_ids[index], grant_request)
+            index = index + 1
+        self.registration_request = registration_request
+
+    def getCbsdId(self):
+        return self.cbsd_id
+
+    def getGrantObject(self, grant_id):
+        return self.grant_objects[grant_id]
+
+    def getRegistrationRequest(self):
+        return self.registration_request
+
+    def doesCbsdHasActiveGrant(self):
+        for grant_object in self.grant_objects.values():
+            return grant_object.isGrantActive()
+
+    def constructHeartbeatRequestForAllActiveGrants(self):
+        """
+        construct list of heartbeat requests of all active grants
+        """
+        heartbeat_requests = []
+        for grant_object in self.grant_objects.values():
+            if(grant_object.isGrantActive()):
+                heartbeat_request = grant_object.constructHeartbeatRequest()
+                heartbeat_requests.append(heartbeat_request)
+        return heartbeat_requests
+
+    def getOperationParamsOfAllAuthorizedGrants(self):
+        """
+        Returns the list of operation params of all authorized grants
+        """
+        operation_params = []
+        for grant_object in self.grant_objects.values():
+            if(grant_object.isGrantAuthorizedInLastHeartbeat()):
+                operation_param = grant_object.getRequestOperationParam()
+                operation_params.append(operation_param)
+        return operation_params
+
+
+class DomainProxy():
+    """
+        Holds the domain proxy related parameters
+    """
+
+    def __init__(self, ssl_cert, ssl_key, testcase):
+        """
+        Args:
+            ssl_cert: Path to SSL cert file
+            ssl_key: Path to SSL key file
+            testcase: test case object from the caller
+        """
+        self.ssl_cert = ssl_cert
+        self.ssl_key = ssl_key
+        self.cbsd_objects = {}
+        self.testcase = testcase
+
+    def initialize(self, registration_requests, grant_requests):
+        """
+        Args
+            registration_requests:  A list of dictionary with a single key-value pair
```

Comment:
Unwrap.

Reply:
As per   suggestion used the non wrapped version of grant requests